next_generation/pint_server/app.py

```diff
@@ -57,33 +58,20 @@ def get_supported_providers():
     global CACHE_PROVIDERS
 
     if CACHE_PROVIDERS is None:
-        CACHE_PROVIDERS = query_providers()
+        versions  = VersionsModel.query.with_entities(VersionsModel.tablename)
+        providers = []
```

Comment:
This could probably be done with a simple set comprehension, converted to a list at the end if you need CACHE_PROVIDERS to be a list, rather than a set... E.g.

 
```suggestion
        providers = list({re.sub('(servers|images)', '', v.tablename)
                          for v in versions})
```

Reply:
We want the values to be unique. Otherwise, we'll end up with two 'amazon' entries for 'amazonservers' and 'amazonimages'.

Reply:
My suggestion used a "set" comprehension, which means that there can be no duplicates in the resulting list, since the set would eliminate them ;-)

Reply:
Let me give it a try.

Reply:
works! Thanks!